aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt

```diff
@@ -440,12 +440,44 @@ open class AztecText : AppCompatEditText, TextWatcher, UnknownHtmlSpan.OnUnknown
     // Setup the keyListener(s) for Backspace and Enter key.
     // Backspace: If listener does return false we remove the style here
     // Enter: Ask the listener if we need to insert or not the char
-    private fun setupBackspaceAndEnterDetection() {
+    private fun setupKeyListenersAndInputFilters() {
         //hardware keyboard
         setOnKeyListener { _, _, event ->
             handleBackspaceAndEnter(event)
         }
 
+        // This InputFilter created only for the purpose of avoiding crash described here:
+        // https://android-review.googlesource.com/c/platform/frameworks/base/+/634929
+        // https://github.com/wordpress-mobile/AztecEditor-Android/issues/729
+        // the rationale behind this workaround is that the specific crash happens only when adding/deleting text right
+        // before an AztecImageSpan, so we detect the specific case and re-create the contents only when that happens.
+        // This is indeed tackling the symptom rather than the actual problem, and should be removed once the real
+        // problem is fixed at the Android OS level as described in the following url
+        // https://android-review.googlesource.com/c/platform/frameworks/base/+/634929
+        val dynamicLayoutCrashPreventer = InputFilter { source, start, end, dest, dstart, dend ->
+            var temp = source
+            if (dest.length > dend+1) {
```

Comment:
Can we move this check ` if (dest.length > dend+1` as first line in the filter, and do not create the `temp` variable since it's not needed in that case?

Reply:
It wasn't actually creating a new variable (the object referenced to by `temp` and `source` where the same), but nevertheless thanks to your comment I looked into the docs again, and made an adjustment in 59ff4c08 as per the docs (initializing temp with `null` by default so the original change gets accepted if we don't need to make any other changes here).

Then we only override it in the case we actually go forward with the text change, by making `temp = ""` in [this other line ](https://github.com/wordpress-mobile/AztecEditor-Android/pull/801/files#diff-7ca71cd1f009ab8438e0901205001b14R481), to actually discard the original change.

Ref: [InputFilter docs](https://developer.android.com/reference/android/text/InputFilter#filter(java.lang.CharSequence,%20int,%20int,%20android.text.Spanned,%20int,%20int))